domain/src/ValueObjects/Transaction.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Domain\ValueObjects;
+
+use Brick\DateTime\LocalDate;
+use Domain\Enums\Operation;
+use Domain\Tests\Factories\ValueObjects\TransactionFactory;
+use Domain\ValueObjects\Exceptions\TransactionException;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Stringable;
+
+final class Transaction implements Stringable
+{
+    use HasFactory;
+
+    /** @throws TransactionException */
+    public function __construct(
+        public readonly LocalDate $date,
+        public readonly Operation $operation,
+        public readonly bool $isIncome,
+        public readonly FiatAmount $costBasis,
+        public readonly ?string $sentAsset,
+        public readonly Quantity $sentQuantity,
+        public readonly bool $sentAssetIsNft,
+        public readonly ?string $receivedAsset,
+        public readonly Quantity $receivedQuantity,
+        public readonly bool $receivedAssetIsNft,
+        public readonly ?string $transactionFeeCurrency,
+        public readonly Quantity $transactionFeeQuantity,
+        public readonly ?string $exchangeFeeCurrency,
+        public readonly Quantity $exchangeFeeQuantity,
+    ) {
+        match ($operation) {
+            Operation::Receive => $this->validateReceive($sentAsset, $receivedAsset, $receivedQuantity),
+            Operation::Send => $this->validateSend($sentAsset, $sentQuantity, $receivedAsset),
+            Operation::Swap => $this->validateSwap($sentAsset, $sentQuantity, $receivedAsset, $receivedQuantity),
+            Operation::Transfer => $this->validateTransfer($sentAsset, $sentQuantity, $receivedAsset),
+        };
```

Comment:
ℹ️ This may change for individual operation value objects down the line, each holding their own validation logic